bin/test-run

```diff
@@ -31,8 +31,36 @@ function check_if_k8s_reachable(){
 }
 
 function run_test(){
-    printf "Running test [%s] %s\\n" "$(basename "$1")" "$2"
-    go test -v "$1" --linkerd="$linkerd_path" --linkerd-namespace="$linkerd_namespace" --k8s-context="$k8s_context" --integration-tests "$2"
+    filename="$1"
+    shift
+
+    printf "Running test [%s] %s\\n" "$(basename "$filename")" "$@"
+    go test -v "$filename" --linkerd="$linkerd_path" --k8s-context="$k8s_context" --integration-tests "$@"
+}
+
+# Install the latest edge release.
+# $1 - namespace to use for the edge release
+function install_edge() {
+    tmp=$(mktemp -d -t l5dbin.XXX)
+    trap "rm -rf $tmp" EXIT
+
+    curl -s https://run.linkerd.io/install-edge | HOME=$tmp sh > /dev/null 2>&1
+
+    local linkerd_path=$tmp/.linkerd2/bin/linkerd
+    local edge_namespace=${1}
+    $linkerd_path install --linkerd-namespace="$edge_namespace" | kubectl apply -f - > /dev/null 2>&1
+    $linkerd_path check --linkerd-namespace="$edge_namespace" > /dev/null 2>&1
+}
+
+function run_upgrade_test() {
+    local edge_namespace="$linkerd_namespace"-upgrade
```

Comment:
with this being the only remaining place where we reference the global `linkerd_namespace`, mind just passing this in a param?

```bash
run_upgrade_test "$linkerd_namespace"-upgrade
```